### ISSUE TITLE
🎙 talk.js - July 2019

### ISSUE THREAD

**Issue by ongmin** [SingaporeJS/talk.js#24]
### 📝 Details
- [x] Event: [talk.js](https://www.meetup.com/Singapore-JS/)
- [x] Date: 10 Jul (Wed)
- [x] Hosts: @secoif
- [x] Venue: HooQ

### 📅 Agenda

Time   | Agenda
------ | :-----
7.00pm | Doors open - _mingle with guests and speakers over 🍕_
7.30pm | Start of event - _opening segment by host_
7.40pm | Start of scheduled talks
8.40pm | Open announcements - _open segment for anyone to make announcements_
9.00pm | End of event

### :speech_balloon: Talks

- [ ] My Awesome Talk by _You_

### :seedling: Propose a talk this month

Want to speak at this month's talk? Leave a comment below with the following details:

<details>
<summary>Click for details</summary>

> #### Title of your topic
> Insert catchy title here
>
> #### Topic description
> Tell us what you'd like to talk about
>
> #### Presenter(s)
> Include GitHub handles where possible. Add your Twitter handle if you wish
>
> #### Presenter's bio
> A short introduction of yourself
>
> #### Duration
> 5 minutes for presenting an introduction to the topic, then up to 20 minutes for discussion.
>
> #### Deck URL (if any)
> You can add this part in later

</details>

### :loudspeaker: Open announcements
Have something you'd like to tell the audience? We have an open announcement segment for you to announce job opportunities, promote a meet up, upcoming conferences, etc.

---

### :pencil: Organizers' checklist

<details>
<summary>Click for details</summary>

- [x] Check available dates on [EngineersSG](https://engineers.sg/)
- [x] Confirm date and venue availability
- [ ] Create event on [SingaporeJS Meetup Page](https://www.meetup.com/singapore-js/)
- [ ] Inform @engineersftw for video recordings
- [ ] Announce event on [SingaporeJS Gitter](https://gitter.im/SingaporeJS/home) and other social media platforms
- [ ] Prepare deck for event

\* _checked boxes indicate confirmed/done_

</details>

**Comment by joeytwiddle** [comment]
*Title of your topic*
Loopback 4 - Building an API with TypeScript

*Topic description*
I will show how to build an API server using the Loopback framework. I will also share some of the gotchas we met with Loopback and TypeScript.

(Loopback will let us connect to multiple databases, validate our inputs, generate an OpenAPI file, and maybe even generate a GraphQL schema.)

*Presenter(s)*
@joeytwiddle

*Presenter's bio*
I work for a company that builds apps using JavaScript.

*Duration*
I guess 25 minutes presentation, then 5 minutes Q+A

*Deck URL (if any)*
https://slides.com/joeytwiddle/loopback4/

**Comment by bendechrai** [comment]
**Title of your topic**
A token walks into a SPA

**Topic description**
Securing SPAs isn't as straight forward as a traditional web application. The code is there for all to see. Cookies and local storage aren't 100% safe. Where do we put the login forms? And what if the user refreshes?!

Learn everything you need to know about JSON Web Tokens, and maybe even some things you don't! Keep user credentials safe, while still communicating effectively with APIs.

**Presenter(s)**
@bendechrai

**Presenter's bio**
Ben Dechrai is a technologist with a staunch focus on security and privacy. This started at the age of 11, when he wrote software to stop his mum from breaking the family PC, and resulted in his working as a developer evangelist for Auth0. He enjoys helping developers find the joy of experimentation, from ethical skulduggery to subversive automation, and can be found on Twitter and Instagram at @bendechrai.

**Duration**
About half an hour

**Comment by omonk** [comment]
**Title of your topic**
Keep your static site up to date without lifting a finger.

**Topic description**
Static sites are great 🎉. They're performant, fast, stable and predictable. But what about data? We don't want to maintain a server or make any runtime requests to external services for fresh, up to date info. We definitely don't want to up date our site manually every time some data changes... So how can we do it? 

In my talk I will cover my experiences of building, deploying and maintaining yld.io. Using an array of services and frameworks we are able to make sure data that is always changing is kept up to date without even realising.

**Presenter(s)**
@omonk

**Presenter's bio**
I'm a JS dev working with React and Node for yld.io 

**Duration**
30mins

**Deck URL (if any)**
working on it!

**Comment by mikhfilippov** [comment]
Title of your topic
What Babylonian Astronomers Could Learn About Javascript?  

Topic description
Software engineering collaboration is hard. Software engineers spend > 70% of their time learning about their own team’s source code. Enormous size, constant change and intricate dependencies of the source code - are the main reason causing that problem. 

To help software engineers at Quod AI we translate source code into natural language to make it easier to search, navigate and understand. In order to do that we use neural network models, natural language processing algorithms and statistical models. 

In this talk we will share some of the insights that we gained from analyzing more than 300 millions lines of code.

Presenter(s)
https://github.com/mikhfilippov

Presenter's bio
I am an applied mathematician, who is building an AI platform that explains source code in plain English.

Duration
I guess 20 minutes presentation, then 10 minutes Q+A

**Comment by ongmin** [comment]
Hi @mikhfilippov! Thanks for offering to share, and interesting topic about Patterns in JS. :) 

Would it be possible to have your talk in August (14th) instead? I'm looking at the previous 3 talk submissions and they are all long form talks (20-30minutes), and it would be great to not rush them. Let me know what you think.

August's meetup: https://github.com/SingaporeJS/talk.js/issues/25

**Comment by mikhfilippov** [comment]
Hi @ongmin ! Sure, no problems. I submitted the talk for August.

**Comment by ongmin** [comment]
Hi @bendechrai @joeytwiddle @omonk, 
we're all set for July, can we work with the timings below?

1) Loopback 4 - Building an API with TypeScript - 25 mins + 5 mins Q&A
2) A token walks into a SPA - 25 mins + 5 mins Q&A
3) How to keep your static site up to date without lifting a finger - 20 mins + 5 mins Q&A

Thanks and see you next Wednesday.

**Comment by bendechrai** [comment]
25 minutes should be good. I can take questions at he end of the evening too if it comes to that :)

See you next week!

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by omonk** [comment]
Sounds good to me!

**Comment by ongmin** [comment]
Thanks all for attending! Feel free to share your slides here before I close the issue. :) Unfortunately our usual Engineers.SG super volunteer was busy and we couldn’t do recordings this month. 

Thanks again for sharing your talks!
